docs/migration-guide-js.md

```diff
@@ -0,0 +1,138 @@
+const parser = require('raml-1-parser')
+
+async someFunc () {
+  const api = await parser.loadRAML('/path/to/raml10/api.raml')
+  const expanded = api.expand(true)
+  // Do something with the expanded API
+}
+
+someFunc()
+```
+
+Here's how it can be reworked to use `webapi-parser`:
+```js
+const wap = require('webapi-parser').WebApiParser
+
+async someFunc () {
+  // Or wap.raml08 for RAML 0.8 operations
+  const api = await wap.raml10.parse('file:///path/to/raml10/api.raml')
+  const expanded = await wap.raml10.resolve(api)
+  // Do something with the expanded API
+}
+
+someFunc()
+```
+
+In the example above, object `WebApiParser` contains namespaces for all the supported API Spec formats: `raml10`. `raml08`, `oas20`, `oas30`, `amfGraph`, each having an identical interface (OAS namespaces have an extra few methods). The list of supported operations each format supports includes parsing, resolution(flattening), validation and string generation.
+
+To get a description of each namespace and operation please research the new model API by following the link in the [assistance section](#need-assistance) below.
+
+## Detailed migration examples
+This section lists migration examples of the most common `raml-1-parser` parsing and model methods. Snippets are separated with a newline. First line of each example shows `raml-1-parser` method usage, while the second line shows how to achieve the same functionality with `webapi-parser` if possible. If no obvious alternative exists, a comment gives more detail.
+
+### Parsers
+```js
+const oldParser = require('raml-1-parser')
+const wp = require('webapi-parser')
+const wap = wp.WebApiParser
+
+const node = await oldParser.loadApi('/path/to/api.raml')
+const model = await wap.raml10.parse('file:///path/to/api.raml')
+// or
+// const model = wap.raml08.parse('file:///path/to/api.raml')
+
+oldParser.asFragment(node)
+// Not necessary. Webapi-parser parses fragments into different model types.
+
+oldParser.isFragment(node)
+!(model instanceof wp.webapi.WebApiDocument)
+
+oldParser.loadApiSync('/path/to/api.raml')
+// Not supported
+
+oldParser.loadRAML('/path/to/api.raml')
+wap.raml10.parse('file:///path/to/api.raml')
+wap.raml08.parse('file:///path/to/api.raml')
+
+oldParser.loadRAMLSync('/path/to/api.raml')
+// Not supported
+
+oldParser.parseRAML('#%RAML 1.0\n...')
+wap.raml10.parse('#%RAML 1.0\n...')
+wap.raml08.parse('#%RAML 1.0\n...')
+
+oldParser.parseRAMLSync('#%RAML 1.0\n...')
+// Not supported
+```
+
+### API Models
+```js
+const oldParser = require('raml-1-parser')
+const wp = require('webapi-parser')
+const wap = wp.WebApiParser
+
+const node = await oldParser.loadApi('/path/to/api.raml')
+const model = await wap.raml10.parse('file:///path/to/api.raml')
+// or
+// const model = wap.raml08.parse('file:///path/to/api.raml')
+
+node.resources()
+model.encodes.endPoints()
+// Note that webapi-parser resources are flat and occur in the order defined in the RAML doc.
+
+node.resources()[0].methods()
```

Comment:
add a short description of what it does before each block

Reply:
Added